src/libraries/System.Private.CoreLib/src/System/Resources/ManifestBasedResourceGroveler.cs

```diff
@@ -307,6 +281,50 @@ internal ResourceSet CreateResourceSet(Stream store, Assembly assembly)
             }
         }
 
+        [RequiresUnreferencedCode("This Path uses Reflection to activate types on an external stream.")]
```

Comment:
Have you tried trimming an app with the `AllowReflectionForNonPrimitiveTypes=true` turned on? This is the ILLink warning message that gets shown to the user right?

If so, we should update this message to be more user friendly. It should read more like what we have in the `Strings.resx` file.

See 
https://github.com/dotnet/runtime/blob/9f759f2975722071d844d0157402707c9bd2f010/src/coreclr/System.Private.CoreLib/src/System/StartupHookProvider.cs#L106-L107

For how we word it for startup hooks.

Reply:
Same for the message on `InitializeBinaryFormatter()`.

Will the app developer get 2 ILLink warnings? Is it possible to narrow it down to just 1 good warning?